polars/polars-core/src/frame/mod.rs

```diff
@@ -3328,13 +3328,21 @@ impl DataFrame {
         &self,
         cols: &[String],
         stable: bool,
+        include_key: bool,
     ) -> PolarsResult<Vec<DataFrame>> {
         let groups = if stable {
             self.groupby_stable(cols)?.take_groups()
         } else {
             self.groupby(cols)?.take_groups()
         };
 
+        // drop key columns prior to calculation if requested
+        let df = if !include_key {
```

Comment:
nit: No need to negate. Simply swap the expressions. That reads a little easier.

Reply:
Good catch. fixed.